### ISSUE TITLE
add basic Elixir project with simple span creation and use of Otel Exporter

### ISSUE THREAD

**Issue by tsloughter** [opentelemetry-beam/opentelemetry-examples#5]
No plug, no phoenix, nothing like that, just an Elixir project that has a single application that exports a function you can call that starts a couple spans.

The configuration should use the `opentelemetry_exporter` and come with a docker compose setup to run the collector, zipkin and jaeger that it sends spans to both of them.

Probably good to also either have a separate project or a separate config in this example that uses the zipkin exporter directly to export to both zipkin and jaeger.

**Comment by kw7oe** [comment]
Hi Tristan, I am interested to tackle this issue and have setup a simple project [here](https://github.com/kw7oe/elixir-opentelemetry-example) that have:

- docker compose to setup the otel collector, zipkin, jaeger and the elixir application _(tested in MacOS only)_
- elixir application with `opentelemetry-api` and `opentelemetry-exporter` as dependencies and simply called `OpenTelemetry.Tracer.with_span/2` and `OpenTelemetry.Span.set_attribute/2`, so there is only one trace currently. 

Do let me know any recommended improvements needed on the sample project. 

And I am also wondering what's the pathway to include the example into this repository. Here are some questions I have in mind:

- Do we create a separate folder with in this repository, where a separate docker compose is placed in that folder?
- What are the prefer naming of the folder for this sample? `elixir-example`?

I also notice that [`opentelemetry-erlang-api`](https://github.com/open-telemetry/opentelemetry-erlang-api) will be deprecated sometime in the future. So, I'm not sure would we want to include the example after the deprecation or is fine to just proceed with what we currently have.

And also thank you and the Erlang Ecosystem Foundation Observability Working Group members for the works on making OpenTelemetry available in the Erlang/Elixir ecosystem 🙏

**Comment by tsloughter** [comment]
@kw7oe nice!

Regarding the API, could you for the time being change the example to reference the app in the repo using mix's `sparse_git` feature? This way it'll be the latest and then can update it to the hex package when we get a new release made in the coming weeks. So something like:

```
{:opentelemetry_api, github: "open-telemetry/opentelemetry-erlang", sparse:
"apps/opentelemetry_api"}
```

**Comment by kw7oe** [comment]
Sure @tsloughter, I'll open a PR soon with the recommended changes 💯 

Then, we can resume any recommended changes over the PR.

**Comment by GregMefford** [comment]
I think there's still some improvements we can make so that this example (and the others) are more useful / interesting, but this issue has been resolved now that #6 is merged.
